chef-config/lib/chef-config/mixin/train_transport.rb

```diff
@@ -116,6 +116,10 @@ def build_transport
         train_config = tm_config.to_hash.select { |k| Train.options(protocol).key?(k) }
         logger.trace("Using target mode options from #{ChefUtils::Dist::Infra::PRODUCT} config file: #{train_config.keys.join(", ")}") if train_config
 
+        unless credentials.slice(:user, :train_user).values.include? "root"
```

Comment:
In fact, only the root user (with all permissions) is supported right now and only Linux/FreeBSD. It's a warning only though 😅

I changed to check first `:train_user` and then `:user`, because the first is an override for Kitchen EC2's inside workings, while the second would still work with Chef Infra standalone. 

The related code for the need of `train_user`is https://github.com/test-kitchen/kitchen-ec2/blob/main/lib/kitchen/driver/ec2.rb#L377-L390

Reply:
@jaymzh I don't know if @thheinen's comment resolves this for you or not?

Reply:
That seems... Better. 

But this whole thing is originally written to support different platforms, so it seems worthwhile to try to make this a bit more future proof. What if you're using sudo on the other side? Or as previously mentioned we want to support switches, embedded Windows, etc etc.

I understand it is only a warning, but lots of warnings that aren't real problems is what makes people ignore warnings. I think we should probably add a handful of known admin-y usernames to at least minimize the false positives.

That said, my main concern is addressed, so if progress wants to move forward, their call. But it seems like minimal effort to make this more future proof.

Reply:
Totally get those arguments.

The thing is that Target Mode reuses the shell-commands from resources' non-TM implementations (so no `sudo` in there) + the file transfers via Train are used (and afaik there's no "sudo-on-scp". plus, if we want to keep this open for other backends like WinRM, ...).

We can check after 19.0 if it's an option to make TM check root-ism and magically add sudo if needed + how to fix the file transfers. There'll be work to extend scope after anyway. Should definitely put that as a feature request!